### PR TITLE
FileUploaderField component, refs erm-337

### PR DESCRIPTION
**Pull request by aditya-matukumalli**
- Branched out the _FileUploaderField_ component out of the _DocumensFieldArray_ erm-component to be used for creating a new kb-import functionality.

### CODE REVIEW

File: lib/FileUploaderField/tests/interactor.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@interactor class FileUploaderFieldInteractor {
+  static defaultScope = '[data-test-document-field-file]';
+
+  hasFilename = isPresent('[data-test-fuf-name]');
+  hasUploaded = isPresent('[data-test-fuf-uploaded]');
+  hasDeleteBtn = isPresent('[data-test-fuf-delete]');
+
+  filename = text('[data-test-fuf-name]');
+  uploaded = text('[data-test-fuf-uploaded]');
+  delete = clickable('[data-test-fuf-delete]');
```
Comment (doytch) [inline lib/FileUploaderField/tests/interactor.js]: [`delete` is a reserved keyword in JS.](https://ecma-international.org/ecma-262/6.0/#sec-reserved-words) I guess this didn't throw any errors beforehand but we should take the chance to give it a better name. (Pretty sure I wrote this line originally. 🤦‍♂ )

File: lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js [hunk 2]
```diff
@@ -175,151 +172,6 @@ describe('DocumentsFieldArray', () => {
     });
   });
 
-  describe('attaching a file', () => {
```
Comment (doytch) [inline lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js]: Can you keep a basic test that just ensures that the file uploader is present [if `onUploadFile` is provided to DFA](https://github.com/folio-org/stripes-erm-components/blob/e0119dd000d98173c4ff3d119f6cd86116e4b38c/lib/DocumentsFieldArray/DocumentsFieldArray.js#L143)?

**Comment by doytch** [comment]
Also, any idea [why these lines aren't being executed](https://jenkins-aws.indexdata.com/job/folio-org/job/stripes-erm-components/job/ERM-337-fileUploaderField/3/LCov_20Coverage_20Report/lib/FileUploaderField/FileUploaderFieldView.js.html#L43)? 🤔

File: lib/FileUploaderField/tests/interactor.js (hunk 1 above)
Comment (aditya-matukumalli) [reply lib/FileUploaderField/tests/interactor.js]: This line https://github.com/folio-org/stripes-erm-components/blob/e0119dd000d98173c4ff3d119f6cd86116e4b38c/lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js#L89 should take care of it? shouldn't it? unless you are referring to something else here?

File: lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js (hunk 2 above)
Comment (aditya-matukumalli) [reply lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js]: This line https://github.com/folio-org/stripes-erm-components/blob/e0119dd000d98173c4ff3d119f6cd86116e4b38c/lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js#L89 should take care of it? shouldn't it? unless you are referring to something else here?

**Comment by aditya-matukumalli** [comment]
> Also, any idea [why these lines aren't being executed](https://jenkins-aws.indexdata.com/job/folio-org/job/stripes-erm-components/job/ERM-337-fileUploaderField/3/LCov_20Coverage_20Report/lib/FileUploaderField/FileUploaderFieldView.js.html#L43)? 🤔

There was an _expect_ instead of an _it_ inside the describe block. Corrected that.

File: lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js (hunk 2 above)
Comment (doytch) [reply lib/DocumentsFieldArray/tests/DocumentsFieldArray-test.js]: Ahh yeah, true. I didn't scroll up enough.

**Pull request merged**